storage/innobase/handler/handler0alter.cc

```diff
@@ -3807,7 +3807,16 @@ static void dd_commit_inplace_alter_table(
                                  FTS_DOC_ID_INDEX_NAME, col);
     }
 
-    dd_space_id = dd_first_index(old_dd_tab)->tablespace_id();
+    /* This can happen only with expanded fast index creation. On the
+    intermediate table during ALTER COPY, we drop secondary indexes using
+    inplace alter APIs. The old definition here is old copy of table. Hence we
+    should use new_dd_tab here for updating the dd::Indexes */
+    if (new_table->skip_alter_undo) {
```

Comment:
Maybe
`dd_space_id = dd_first_index(new_table->skip_alter_undo ? new_dd_tab : old_dd_tab)->tablespace_id();`
?

Reply:
Dont like the readability of this suggestion.

Reply:
Fair enough